components_library/boards/abstract_board.py

```diff
@@ -167,6 +165,26 @@ def set_mem_mode(self, mem_mode: MemMode) -> None:
         """
         raise NotImplementedError
 
+    def setup_memory_ranges(self) -> None:
+        """
+        Set the memory ranges for this board.
+
+        This is called by `connect_things`. It can query the board's memory
+        to determine the size and the set the memory ranges on the memory if
+        it needs to move the memory devices.
+
+        The simplest implementation just sets the board's memory range to be
+        the size of memory and memory's memory range to be the same as the
+        board. Full system implementations will likely need something more
+        complicated.
+        """
+        memory = self.get_memory()
+
+        # The simple board just has one memory range that is the size of the
+        # memory.
+        self.mem_ranges = [AddrRange(memory.get_size())]
+        memory.set_memory_range(self.mem_ranges)
```

Comment:
I think, given your documentation here, you meant this to go in the Simple board, and this function to be abstract? That's how i'd do it.

Reply:
Wait... why would this go in simple board and connect_things go in abstract? They should both be in one or the other.